### ISSUE TITLE
EqualsAvoidNull - Support null check

### ISSUE THREAD

**Issue by nkavian** [checkstyle/checkstyle#10638]
**How it works Now:**

My config: `<module name="EqualsAvoidNull"/>`

My code: `if (criteria == null || criteria.isEmpty() || criteria.equals("{}")) {`

The checkstyle message: `String literal expressions should be on the left side of an equals comparison. [EqualsAvoidNull]`

**Is your feature request related to a problem? Please describe.**

The code is more readable with the line of code I've shown above.

**Describe the solution you'd like**

Instead of suppressing the warning, I'd like to request a feature request.
I'd like `EqualsAvoidNull` to detect that the object is not null and not show me this warning.

**Comment by pbludov** [comment]
From the check [description](https://checkstyle.org/config_coding.html#EqualsAvoidNull):

 > Checks that any combination of String literals is on the left side of an equals() comparison.

It does not matter if the object `null` or not. This check is about to use `"{}".equals(criteria)` instead of `criteria.equals("{}")`.

I understand correctly what you are coding like

```java
if (criteria == null || criteria.equals("{}")) {  // null check is here
}

if ("{}".equals(otherCriteria)) {  // no null check, so we need to use different code style
}
```
?

**Comment by nkavian** [comment]
In my example, there are 3 separate conditions, and they can not be combined or removed. Your example removes conditions.

I understand how the rule works. I wrote the issue to ask for the rule to be intelligent to avoid forcing a developer to rewrite the code like this:

```
if (criteria == null || criteria.isEmpty() || "{}".equals(criteria)) {
```

This is less readable compared to:

```
if (criteria == null || criteria.isEmpty() || criteria.equals("{}")) {
```

**Comment by Neha8482** [comment]
I have a solution
